tests/integration/README.md

```diff
@@ -1,21 +1,161 @@
+
+### Valid Test Definitions
+
+The most basic test you can provide can be written 4 ways since `cmds` and `conditions` can both be 2 different types:
+
+    "cmds as string, conditions as Condition": {
+        "cmds": "echo hello",
+        "conditions": HasReturnCode(),
+    }
+
+    "cmds as list, conditions as Condition": {
+        "cmds": ["echo hello"],
+        "conditions": HasReturnCode(),
+    }
+
+    "cmds as string, conditions as list": {
+        "cmds": "echo hello",
+        "conditions": [HasReturnCode()],
+    }
+
+    "cmds as list, conditions as list": {
+        "cmds": ["echo hello"],
+        "conditions": [HasReturnCode()],
+    }
+
+Adding slightly more complexity, we provide a run type:
+
+    "basic test with run type": {
+        "cmds": "echo hello",
+        "conditions": HasReturnCode(),
+        "run type": "local" # This could also be "distributed"
+    }
+
+Now we'll add a cleanup command:
+
+    "basic test with cleanup": {
+        "cmds": "mkdir output_dir/",
+        "conditions": HasReturnCode(),
+        "run type": "local",
+        "cleanup": "rm -rf output_dir/"
+    }
+
+Finally we'll play with the number of processes to start:
+
+    "test with 1 process": {
+        "cmds": "mkdir output_dir/",
+        "conditions": HasReturnCode(),
+        "run type": "local",
+        "cleanup": "rm -rf output_dir/",
+        "num procs": 1
+    }
+
+    "test with 2 processes": {
+        "cmds": ["mkdir output_dir/", "touch output_dir/new_file.txt"],
+        "conditions": HasReturnCode(),
+        "run type": "local",
+        "cleanup": "rm -rf output_dir/",
+        "num procs": 2
+    }
+
+Similarly, the test with 2 processes can be condensed to a test with 1 process
+by placing them in the same string and separating them with a semi-colon:
+
+    "condensing test with 2 processes into 1 process": {
+        "cmds": ["mkdir output_dir/ ; touch output_dir/new_file.txt"],
+        "conditions": HasReturnCode(),
+        "run type": "local",
+        "cleanup": "rm -rf output_dir/",
+        "num procs": 1
+    }
+
+
+### Invalid Test Definitions
+
+No `cmds` provided:
+
+    "no cmd": {
+        "conditions": HasReturnCode(),
+    }
+
+No `conditions` provided:
+
+    "no conditions": {
+        "cmds": "echo hello",
+    }
+
+Number of `cmds` does not match `num procs`:
+
+    "num cmds != num procs": {
+        "cmds": ["echo hello; echo goodbye"],
+        "conditions": HasReturnCode(),
+        "num procs": 2
+    }
+
+Note: Technically 2 commands were provided here ("echo hello" and "echo goodbye")
+but since they were placed in one string it will be viewed as one command. 
+Changing the `cmds` section here to be:
+
+    "cmds": ["echo hello", "echo goodbye"]
+
+would fix this issue and create a valid test definition.
 
 
-### Continuous integration
+## Continuous Integration
 Currently run from [Bamboo](https://lc.llnl.gov/bamboo/chain/admin/config/defaultStages.action?buildKey=MLSI-TES).
```

Comment:
This is defunct, bamboo doesn't exist anymore. We should replace this with the github CI info.